### DISCUSSION TITLE
Firefox/Safari Tests in CI are 5x-7x slower

### DISCUSSION THREAD

**Discussion by uuf6429** [php-webdriver/php-webdriver#1118] [General]
Just something I noticed:
![image](https://github.com/user-attachments/assets/30128857-e490-41c5-8f07-e3e09926a5bc)

I guess it doesn't cause any real problems and AFAIK it doesn't affect end users, but it still looks funny. While working on my PR for a moment I was worried that I broke something, then I saw that earlier tests were the same.

**Comment by ernstki** [comment]
There are so many layers involved here, it's hard to say which layer is at fault for the slowdown, but it seems unlikely that the PHP WebDriver library has anything to do with this, doesn't it?

Did you continue to see this problem, reproducibly? Did you still see it with later releases of the webdriver (binaries) themselves, Firefox, Safari, and this library?

Imagine! There's all these bits in play:

* your CI/CD platform, which, if on a shared host, is subject to outages and heavy load from other users
* possibly a Selenium server, running on the JVM, possibly running inside Docker, so you'd have a whole tiny Linux distro's worth of performance problems that could crop up there
* a webdriver binary, like `geckodriver`, which is itself a complex codebase and has hundreds of transitive dependencies, all of which routinely get updates that both improve and degrade performance
    * just as an example: https://github.com/zip-rs/zip2/issues/274
* a WebDriver protocol, the specifications of which change from time to time, requiring webdrivers and client libraries to deal with deprecations and new code paths
* the WebDriver client library that you use to automate your tests, _e.g._ php-webdriver
* a web browser

**Reply by uuf6429** [reply]
Sorry, I wasn't very clear: I meant CI of this project, not mine.

The problem is still happening by the way (latest test run):
https://github.com/php-webdriver/php-webdriver/actions/runs/14234149490

<img src="https://github.com/user-attachments/assets/3759da7e-3a73-4910-8985-9f5343d42f16" width="300"/>

**Reply by ernstki** [reply]
Ha, shows you how often I use GitHub in dark mode, that I didn't even recognize it!

Unfortunate reality is Chrome is eating the world, and maybe that project's webdriver is just better optimized? ¯\\\_(ツ)_/¯
